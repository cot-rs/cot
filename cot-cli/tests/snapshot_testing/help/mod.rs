use super::*;

#[test]
fn no_args() {
    assert_cmd_snapshot!(cot_cli!());
}

#[test]
fn short_help() {
    assert_cmd_snapshot!(cot_cli!("-h"));
}

#[test]
fn long_help() {
    assert_cmd_snapshot!(cot_cli!("--help"));
}

#[test]
fn version() {
    assert_cmd_snapshot!(cot_cli!("--version"));
}

#[test]
fn help() {
    assert_cmd_snapshot!(cot_cli!("help"));
}

#[test]
fn help_new() {
    assert_cmd_snapshot!(cot_cli!("help", "new"));
}

#[test]
fn help_migration() {
    assert_cmd_snapshot!(cot_cli!("help", "migration"));
}

#[test]
fn help_migration_list() {
    assert_cmd_snapshot!(cot_cli!("help", "migration", "list"));
}

#[test]
fn help_migration_make() {
    assert_cmd_snapshot!(cot_cli!("help", "migration", "make"));
}

#[test]
fn help_cli_manpages() {
    assert_cmd_snapshot!(cot_cli!("help", "cli", "manpages"));
}

#[test]
fn help_cli_completions() {
    assert_cmd_snapshot!(cot_cli!("help", "cli", "completions"));
}
