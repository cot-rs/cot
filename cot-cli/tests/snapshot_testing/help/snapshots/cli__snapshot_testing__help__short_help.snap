---
source: cot-cli/tests/snapshot_testing/help/mod.rs
info:
  program: cot
  args:
    - "-h"
---
success: true
exit_code: 0
----- stdout -----
The Rust web framework for lazy developers - CLI tool.

Usage: cot [OPTIONS] <COMMAND>

Commands:
  new        Create a new Cot project
  migration  Manage migrations for a Cot project
  cli        Manage Cot CLI
  help       Print this message or the help of the given subcommand(s)

Options:
  -v, --verbose...  Increase logging verbosity
  -q, --quiet...    Decrease logging verbosity
  -h, --help        Print help
  -V, --version     Print version

----- stderr -----
